projects/plugins/crm/src/js/components/automations-admin/components/edit-modal/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button } from '@automattic/jetpack-components';
+import { Modal } from '@wordpress/components';
+import { sprintf, __ } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import { StepConfig } from '../step-config';
+import styles from './styles.module.scss';
+import type { Step, Workflow } from 'crm/state/automations-admin/types';
+
+type EditModalProps = {
+	workflow: Workflow;
+	isOpen: boolean;
+	onClose: () => void;
+};
+
+export const EditModal: React.FC< EditModalProps > = ( { isOpen, onClose, workflow } ) => {
+	const steps: Step[] = [];
+
+	let stepId: string | undefined = workflow.initial_step;
+	while ( stepId ) {
+		const step: Step = workflow.steps?.[ stepId ];
+		steps.push( step );
+		stepId = step.next_step;
+	}
+
+	const onSave = useCallback( () => {
+		// TODO: save workflow here
+		onClose();
+	}, [ onClose ] );
+
+	const onCancel = useCallback( () => {
+		// TODO: reload workflow here
+		onClose();
+	}, [ onClose ] );
+
+	return (
+		isOpen && (
+			<Modal onRequestClose={ () => onClose() }>
```

Comment:
In Figma the close button seems to be in line with the header. Do you know if Jetpack RNA has any components that could be used to achieve it instead of the WordPress one? E.g. the "Dialog" or "Popover" component?

Glancing through [Storybook](https://automattic.github.io/jetpack-storybook/?path=/story/js-packages-components-popover--default) then it doesn't fit our current needs; if that's the case, then the current implementation is OK

![Screenshot 2023-09-28 at 15 26 48](https://github.com/Automattic/jetpack/assets/3846700/8c479932-d3f8-4e0b-af15-9a0184f5cb60)

Reply:
I have improved this but I'm still not totally happy with the spacing between the header and the subheader. I think that can be an issue to solve later though.